azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsTelemetryAutoConfiguration.java

```diff
@@ -0,0 +1,339 @@
+    public TelemetryClient telemetryClient(TelemetryConfiguration configuration) {
+        return new TelemetryClient(configuration);
+    }
+
+    @Bean
+    public FixedRateSamplingTelemetryProcessor fixedRateSamplingTelemetryProcessor() {
+        Sampling sampling = applicationInsightsProperties.getTelemetryProcessor().getSampling();
+        FixedRateSamplingTelemetryProcessor processor = new FixedRateSamplingTelemetryProcessor();
+        processor.setSamplingPercentage(String.valueOf(sampling.getPercentage()));
+        for (TelemetryType include : sampling.getInclude()) {
+            processor.addToIncludedType(include.name());
+        }
+        for (TelemetryType exclude : sampling.getExclude()) {
+            processor.addToExcludedType(exclude.name());
+        }
+        return processor;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public TelemetryChannel telemetryChannel() {
+        InProcess inProcess = applicationInsightsProperties.getChannel().getInProcess();
+        return new InProcessTelemetryChannel(inProcess.getEndpointAddress(),
+                String.valueOf(inProcess.getMaxTransmissionStorageFilesCapacityInMb()), inProcess.isDeveloperMode(),
+                inProcess.getMaxTelemetryBufferCapacity(), inProcess.getFlushIntervalInSeconds(), inProcess.isThrottling(),
+            inProcess.getMaxInstantRetry());
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "azure.application-insights.quick-pulse.enabled", havingValue = "true", matchIfMissing = true)
+    @DependsOn("telemetryConfiguration")
+    public QuickPulse quickPulse() {
+        QuickPulse.INSTANCE.initialize();
+        return QuickPulse.INSTANCE;
+    }
+
+    @Bean
+    public InternalLogger internalLogger() {
+        Map<String, String> loggerParameters = new HashMap<>();
+        ApplicationInsightsProperties.Logger logger = applicationInsightsProperties.getLogger();
+        loggerParameters.put("Level", logger.getLevel().name());
+        InternalLogger.INSTANCE.initialize(logger.getType().name(), loggerParameters);
+        return InternalLogger.INSTANCE;
+    }
+
+    @Bean
+    public PerformanceCounterContainer performanceCounterContainer() {
+        ApplicationInsightsProperties.PerformanceCounter performanceCounter = applicationInsightsProperties.getPerformanceCounter();
+        PerformanceCounterContainer.INSTANCE.setCollectionFrequencyInSec(performanceCounter.getCollectionFrequencyInSeconds());
+
+        ApplicationInsightsProperties.Jmx jmx = applicationInsightsProperties.getJmx();
+        if (jmx.getJmxCounters() !=null && jmx.getJmxCounters().size() > 0) {
+            processAndLoadJmxCounters(jmx.getJmxCounters());
+        }
+        return PerformanceCounterContainer.INSTANCE;
+    }
+
+    /**
+     * This method is used to process and load list of JmxCounters provided in the configuration.
+     * @param jmxCounterList
+     */
+    private void processAndLoadJmxCounters(List<String> jmxCounterList) {
```

Comment:
I think it would be better to move all this logic below into `ApplicationInsightsProperties`, to keep auto-configuration class simpler.

Reply:
Yes this refactoring can be done.

Reply:
Done